### PR TITLE
Flood risk widget

### PR DESCRIPTION
**Pull request by afiore**
Abstract out the demographics statics component into a shared function, and uses it to display a tabular view of risk data. Also gracefully handle locations which are not within a risk area

screenshot from 2016-02-24 12 52 09

### CODE REVIEW

**Comment by ostapneko** [comment]
I suppose you're going to update the title once the new version of the app (with the area ID) is deployed?

File: widgets/shared/domUtils.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+function resolveKey(data, keys) {
+  var k = keys[0];
+  if (typeof data[k] === 'object' && keys.length > 1) {
+    return resolveKey(data[k], keys.splice(1));
+  } else {
+    return data[k];
+  }
+}
+
+function dataTable(data, fields) {
```
Comment (ostapneko) [inline widgets/shared/domUtils.js]: I'd rather have a react component

**Comment by afiore** [comment]
What do you think? should we explicitly include that field in widget? I included it in the JSON response just as I wanted to get some sort of audit trail

File: widgets/shared/domUtils.js (hunk 1 above)
Comment (afiore) [inline widgets/shared/domUtils.js]: Ok, sounds better indeed

**Comment by ostapneko** [comment]
I don't know, but I'd rather not have a lat lon displayed in the widget

File: widgets/flood/results.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  render: function() {
+    var data      = this.props.data;
+    var style     = this.props.show ? {} : { display: 'none' };
+    var floodData = data && data.floodRisk;
+    var isAtRisk  = data && data.isAtRisk;
+    var sectionEl;
+
+    if (!data) { return D.div(null); }
+
+    if (config.mapboxMapId && isAtRisk) {
```
Comment (ostapneko) [inline widgets/flood/results.js]: I would add a guard on `floodData`, which could be null and lead to a lot of trouble

File: widgets/flood/results.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      sectionEl = D.section(null,
+        R.createElement(MapboxMap, { geoJSON: floodData.geometry, size: 'half'}),
+        R.createElement(Description, { floodData: floodData, isAtRisk: isAtRisk, size: 'half' })
+      );
+    } else {
```
Comment (ostapneko) [inline widgets/flood/results.js]: Could we check that `isAtRisk` is false, and not `null` or `undefined`?

**Comment by ostapneko** [comment]
Wanted to add guards on the data, but it's an uphill battle with JS anyway.
LGTM, but I'll wait to test it against the prod api

File: widgets/flood/description.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  propTypes: {
+    floodData: R.PropTypes.object,
+    isAtRisk: R.PropTypes.bool
+  },
+
+  render: function() {
+    var fields = [
+      ['dataPublishedOn', 'Data published'],
+      ['likelihood.riskCategory', 'Risk category'],
+      ['suitability.scale', 'Suitability scale']
```
Comment (ostapneko) [inline widgets/flood/description.js]: Wouldn't `suitability.note` be more appropriate? `scale` is not really meant to be human-readable, e.g. `TownToStreet`

**Comment by ostapneko** [comment]
Make sure to update the version so we can publish it after

**Comment by ostapneko** [comment]
The README needs to be updated as well

File: widgets/flood/description.js (hunk 4 above)
Comment (afiore) [inline widgets/flood/description.js]: Ok, how about this?
screenshot from 2016-02-24 18 53 07

**Pull request merged**